code/test/general.js

```diff
@@ -0,0 +1,2306 @@
+    snapshotId = await TestUtil.snapshot()
+    await TestUtil.init()
+  })
+  afterEach(async function () {
+    await TestUtil.revert(snapshotId)
+  })
+
+  // ====== TRACK ======
+  // Test tacking of an ERC20 token
+  // Expected result the token is now tracked
+  it('OPERATION 0 TRACK: must be able to track tokens', async () => {
+    // create wallets and token contracts used througout the tests
+    let alice = await TestUtil.makeWallet('TG-OP0-1', accounts[0], EFFECTIVE_TIME, DURATION)
+    let aliceOldState = await TestUtil.getONEWalletState(alice.wallet)
+    let aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    const { testerc20 } = await TestUtil.makeTokens(accounts[0])
+    let aliceBalanceERC20
+    let aliceWalletBalanceERC20
+    // transfer ERC20 tokens from accounts[0] (which owns the tokens) to alices wallet
+    await testerc20.transfer(alice.wallet.address, 1000, { from: accounts[0] })
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    aliceBalanceERC20 = await testerc20.balanceOf(alice.wallet.address)
+    assert.equal(1000, aliceBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet succesful')
+    aliceWalletBalanceERC20 = await alice.wallet.getBalance(ONEConstants.TokenType.ERC20, testerc20.address, 0)
+    assert.equal(1000, aliceWalletBalanceERC20, 'Transfer of 1000 ERC20 tokens to alice.wallet checked via wallet succesful')
+    console.log('Funded ERC20')
+
+    // Begin Tests
+    let testTime = Date.now()
+
+    testTime = await TestUtil.bumpTestTime(testTime, 60)
+    await TestUtil.transactionExecute(
+      {
+        ...ONEConstants.NullOperationParams, // Default all fields to Null values than override
+        wallet: alice,
+        operationType: ONEConstants.OperationType.TRACK,
+        tokenType: ONEConstants.TokenType.ERC20,
+        contractAddress: testerc20.address,
+        testTime
+      }
+    )
+    // Update alice current State
+    aliceCurrentState = await TestUtil.getONEWalletState(alice.wallet)
+    // Alice Items that have changed - nonce, lastOperationTime, commits, trackedTokens
+    // nonce
+    let nonce = await alice.wallet.getNonce()
+    assert.notEqual(nonce, aliceOldState.nonce, 'alice wallet.nonce should have been changed')
+    assert.equal(nonce.toNumber(), aliceOldState.nonce + 1, 'alice wallet.nonce should have been changed')
+    aliceOldState.nonce = nonce.toNumber()
+    // lastOperationTime
+    let lastOperationTime = await alice.wallet.lastOperationTime()
+    assert.notStrictEqual(lastOperationTime, aliceOldState.lastOperationTime, 'alice wallet.lastOperationTime should have been updated')
+    aliceOldState.lastOperationTime = lastOperationTime.toNumber()
+    // commits
+    let allCommits = await alice.wallet.getAllCommits()
+    assert.notDeepEqual(allCommits, aliceOldState.allCommits, 'alice wallet.allCommits should have been updated')
+    aliceOldState.allCommits = allCommits
```

Comment:
I think checks for operation time, nonce, commits can be put inside one util function

Reply:
Done now in validateUpdateTransaction 🙏